### PR TITLE
Feature add error message for incorrect decorators

### PR DESCRIPTION
**Pull request by sbearrows**
First pass for adding this error message. Error message could prolly still be more informative but let me know what you think!

### CODE REVIEW

File: R/register.R [hunk 1]
```diff
@@ -136,6 +136,18 @@ get_registered_functions <- function(decorations, tag, quiet = FALSE) {
   }
 
   out <- decorations[decorations$decoration == tag, ]
+
+  if(any(!decorations$decoration %in% c("cpp11::register", "cpp11::init"))) {
```
Comment (jimhester) [inline R/register.R:140]: maybe store the result in a variable to make this easier to read?

e.g. 

```r
bad_decorations <- !decorations$decoration %in% c("cpp11::register", "cpp11::init")

if (any(bad_decorations)) { # etc..
```

File: R/register.R [hunk 2]
```diff
@@ -136,6 +136,18 @@ get_registered_functions <- function(decorations, tag, quiet = FALSE) {
...
+    lines <- decorations$line[which(!decorations$decoration %in% c("cpp11::register", "cpp11::init"))]
+    if(length(lines) > 1) {
+      stop(call. = FALSE, paste0("Can't capture cpp11 decorators on lines ",
```
Comment (jimhester) [inline R/register.R:143]: Maybe the error should be something like 

```
cpp11 attributes must be either `cpp11::register` or `cpp11::init`:
- Invalid attribute [[cpp11::foobar]] on line 123 in file xyz.
- Invalid attribute [[cpp11::baz]] on line 456 in file abc.
```

File: R/register.R (hunk 1 above)
Comment (sbearrows) [reply R/register.R:140]: Sure!

**Comment by jimhester** [comment]
I think maybe we should pull this check out into a separate `check_valid_attributes()` function, otherwise we would run in twice.

File: R/register.R [hunk 3]
```diff
@@ -136,6 +136,18 @@ get_registered_functions <- function(decorations, tag, quiet = FALSE) {
   }
 
   out <- decorations[decorations$decoration == tag, ]
+
+  if(any(!decorations$decoration %in% c("cpp11::register", "cpp11::init"))) {
+    lines <- decorations$line[which(!decorations$decoration %in% c("cpp11::register", "cpp11::init"))]
```
[the remark this answers is not in the document]
Comment (sbearrows) [reply R/register.R:141]: :thumbsup:

File: R/register.R (hunk 2 above)
Comment (sbearrows) [reply R/register.R:143]: Oh that looks great!
Is the best way to make multi-line error messages by using a combination of `paste()` statements with `sep = "\n"` ? Or would this be a `message()` following by an empty `stop()` ?

Comment (jimhester) [reply R/register.R:143]: You can actually use [glue](https://github.com/tidyverse/glue) to do this a little more nicely, something like this

```r
bad_lines <- glue::glue_collapse(
  glue::glue("- Invalid attribute `{names}` on line {line} in file '{file}'."),
  "\n"
)

msg <- glue::glue("
  cpp11 attributes must be either `cpp11::register` or `cpp11::init`:
  {bad_lines}
  ")
stop(msg, call. = FALSE)
```

Comment (sbearrows) [reply R/register.R:143]: Oh nice!

**Comment by sbearrows** [comment]
fixes #127

**Pull request merged**